### DISCUSSION TITLE
Disconnecting waiting for heartbeat

### DISCUSSION THREAD

**Discussion by atlasifs88** [quickfix-j/quickfixj#592] [General]
![image](https://user-images.githubusercontent.com/79868230/209469685-9784147c-c230-4e66-b8b5-d6e9147814d2.jpeg)

Lately, we are receiving this message in our FIX log. When this happens, we will miss the exchange trade acknowledgment and we won't be able to know if the trade was received, opened, filled/partially filled, or rejected. We will only find out when we received the end of day report from the exchange all the trade executions. It's becoming a nightmare for us performing the manual entries of these executed trades and backroom accounting operations.

What is the best method of not losing the heartbeat or getting disconnected? This disconnect rarely happens but when it does then it's a headache.

**Comment by chrjohn** [comment]
How can you miss FIX messages? On the next logon it should be resent, shouldn't it?
However, it would be good to have some more log in text format before this happens.
Also check #569 for things to check.

**Reply by atlasifs88** [reply]
It's not resending even on the next logon. 

Would it be possible to request from the exchange FIX server the missed message?

**Reply by chrjohn** [reply]
What does your config look like?

**Reply by atlasifs88** [reply]
Will share the configuration...settings is embedded in the java code. 

At the moment we are not using a config file which is bad. We are in a process of migrating using a config file. 

What specific configuration setting would you like us to send?

**Reply by atlasifs88** [reply]
We are using FIX org.quickfixj dependency, which provided all the default FIX configuration in spring boot.


[default]
FileStorePath=data
ConnectionType=initiator

LogonTimeout=1
LogoutTimeout=1

SenderCompID=xxx
TargetCompID=yyy


SocketConnectHost=x.x.x.x

#SocketConnectHost=localhost

StartTime=09:00:00 Asia/Manila
EndTime=16:30:01 Asia/Manila
HeartBtInt=30
ReconnectInterval=30
ResetOnLogon=Y
ResetOnLogout=Y
ResetOnDisconnect=Y
ValidateIncomingMessage=N

CheckLatency=N
MaxLatency=120


[session]
BeginString=FIXT.1.1
DefaultApplVerID=9

**Reply by chrjohn** [reply]
When you have settings like `ResetOnLogon` or the like your sequence numbers will get reset on each logon and hence there cannot be an automated resend and you will miss messages.

**Reply by atlasifs88** [reply]
We will set ResetOnLogon=N . 

Any other suggestion or recommendation to our configuration to ensure no messages will get missed?

**Reply by chrjohn** [reply]
Remove all the Reset options. But your counter party also needs to follow that config. It will not work when only one side resets or only one side doesn't reset.
